addons/website_slides/models/slide_channel.py

```diff
@@ -801,15 +801,33 @@ def toggle_active(self):
     def message_post(self, *, parent_id=False, subtype_id=False, **kwargs):
         """ Temporary workaround to avoid spam. If someone replies on a channel
         through the 'Presentation Published' email, it should be considered as a
-        note as we don't want all channel followers to be notified of this answer. """
+        note as we don't want all channel followers to be notified of this answer.
+        Also make sure that only one review can be posted per course."""
         self.ensure_one()
         if kwargs.get('message_type') == 'comment' and not self.can_review:
             raise AccessError(_('Not enough karma to review'))
         if parent_id:
             parent_message = self.env['mail.message'].sudo().browse(parent_id)
             if parent_message.subtype_id and parent_message.subtype_id == self.env.ref('website_slides.mt_channel_slide_published'):
                 subtype_id = self.env.ref('mail.mt_note').id
-        return super(Channel, self).message_post(parent_id=parent_id, subtype_id=subtype_id, **kwargs)
+        message = super().message_post(parent_id=parent_id, subtype_id=subtype_id, **kwargs)
+        if message.subtype_id == self.env.ref("mail.mt_comment"):
```

Comment:
Ok so elearning officers cannot post twice on channels ? Seems a bit annoying ^^ . Previous code was done mainly to avoid that people post several reviews on frontend. Here you are blocking the whole post flow for all people, which seems annoying to me. It should stay as a frontend protection (and probably improved compared to current situation, but keeping it as it is is probably sufficient currently).

Reply:
I was thinking the same, we might be able to check that only if a rating was posted as well?

Is it to prevent posting the rating twice, or a comment (generally speaking) twice?

Reply:
Or only check for non-internal users?

Reply:
If I remember well: we don't want people to spam on frontend. They can log a review (with or without rating, as I think it is not mandatory (to test), then modify it, but cannot post another review. So it is something really linked to frontend usage.

This is not limited to portal people only, even standard users commenting on frontend should be limited to a single post.

Manual calls to message_post are normally limited to marketing people (other people should not have rights, still not sure as I did not dive into the code since a few years), hence no limitation on message_post itself.

Reply:
The goal of this task is to make "frontend" and "backend" as much the same as possible, there is no motivation (and almost no technical way) to differentiate the two now.

"standard people" can chose to post from either interface, so the distinction makes little sense based only on entry point. We need more reliable condition based on state of records instead.

I think we are better off completely limiting for now (like it is done in this PR), and then open up the possibility for specific people to post twice if necessary (marking group, ...).

Reply:
Since post a review from a slide is not possible without rating (even user doesn't choose a rating, there is a default value), as discussed I changed the method to allow the internal users to post several messages without rating (from backend).

Reply:
``The goal of this task is to make "frontend" and "backend" as much the same as possible, there is no motivation (and almost no technical way) to differentiate the two now.``

That's quite false. Frontend is completely different from backend: not the same people, not the same flows, not the same environment, ... you should differentiate "I implement some user flows" from "technically they can do it manually". Indeed people can even call "message_post" in a script if they want, but that's not really our main role to consider they do it (except for security purposes, ...).

``I think we are better off completely limiting for now (like it is done in this PR), and then open up the possibility for specific people to post twice if necessary (marking group, ...).``

Oh yes, so you really think we should prevent elearning officers from posting on their courses to discuss content, ... ?

``Since post a review from a slide is not possible without rating (even user doesn't choose a rating, there is a default value), as discussed I changed the method to allow the internal users to post several messages without rating (from backend).``

Cool, thanks :)